elastic/crawler
# Bump jruby and java versions

Bump jruby and java versions to `9.4.7.0` and `21` respectively.

Now (pre-release) is the best time to do this as we don't have to be concerned about users and backwards compatibility. Jruby 9.4.x gives us access to ruby 3.x ([list of changes here](https://rubyreferences.github.io/rubychanges/3.0.html)).

The biggest change between ruby versions that impacted the codebase was [positional argument syntax changes](https://rubyreferences.github.io/rubychanges/3.0.html#keyword-arguments-are-now-fully-separated-from-positional-arguments). I think I've fixed all of them.

Because java has robust backwards compatibility, there were no changes required for upgrading the java version.

- Bump jruby to `jruby-9.4.7.0` (latest version, gives us ruby `3.1.x`)
- Bump java to `21` (latest version supported by jruby)
- Remove unused `.jetty-version` and `.nvmrc` files
- Bump dependency versions that broke after jruby bump
- Fix positional arguments
- Update documentation with instructions on how to setup java and jruby versions

This change will require an update to rubocop, but I'm planning to address that in a follow-up PR as I expect an autoformat to have a wide impact.
